stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -960,93 +960,14 @@ function lmul!(A::UnitLowerTriangular, B::StridedVecOrMat)
     B
 end
 
-for (t, tfun) in ((:Adjoint, :adjoint), (:Transpose, :transpose))
-    @eval begin
-        function lmul!(xA::UpperTriangular{<:Any,<:$t}, B::StridedVecOrMat)
-            A = xA.data
-            m, n = size(B, 1), size(B, 2)
-            if m != size(A, 1)
-                throw(DimensionMismatch("right hand side B needs first dimension of size $(size(A,1)), has size $m"))
-            end
-            pA = parent(A)
-            for j = 1:n
-                for i = 1:m
-                    Bij = $tfun(pA[i,i])*B[i,j]
-                    for k = i + 1:m
-                        Bij += $tfun(pA[k,i])*B[k,j]
```

Comment:
I have removed all of these methods because all they do is to spell out the "unpacking" in `getindex`: w.r.t. the original argument `xA` above, `$tfun(pA[k,i])` is nothing but `xA[i,k]`, which is what is written in the generic code above. I wonder if we should force inlining at the call site to make sure that `adjoint(xA.data.parent[k,i]` is inserted in the generic code above. OTOH, even for a rather big benchmark case I was hardly seeing any benefit.

Reply:
The `getindex` for `Adjoint`/`Transpose` has been marked with `@propagate_inbounds`. The generic code above would inline thes `getindex`s and generate the same IR. So I think these methods are safe to remove?

Reply:
I would very much hope so!